Detector/DetSensitive/src/SimpleCalorimeterSDWithBirksLaw.cpp

```diff
@@ -0,0 +1,80 @@
+#include "DetSensitive/SimpleCalorimeterSDWithBirksLaw.h"
+
+// FCCSW
+#include "DetCommon/DetUtils.h"
+
+// DD4hep
+#include "DDG4/Geant4Mapping.h"
+#include "DDG4/Geant4VolumeManager.h"
+
+// CLHEP
+#include "CLHEP/Vector/ThreeVector.h"
+
+// Geant4
+#include "G4SDManager.hh"
+
+namespace det {
+SimpleCalorimeterSDWithBirksLaw::SimpleCalorimeterSDWithBirksLaw(const std::string& aDetectorName,
+					 const std::string& aReadoutName,
+					 const DD4hep::Geometry::Segmentation& aSeg)
+  : G4VSensitiveDetector(aDetectorName), m_calorimeterCollection(nullptr), m_seg(aSeg) {
+  // name of the collection of hits is determined byt the readout name (from XML)
+  collectionName.insert(aReadoutName);
+}
+
+SimpleCalorimeterSDWithBirksLaw::~SimpleCalorimeterSDWithBirksLaw(){}
+
+void SimpleCalorimeterSDWithBirksLaw::Initialize(G4HCofThisEvent* aHitsCollections)
+{
+  // create a collection of hits and add it to G4HCofThisEvent
+  // deleted in ~G4Event
+  m_calorimeterCollection = new G4THitsCollection
+    <DD4hep::Simulation::Geant4CalorimeterHit>(SensitiveDetectorName,collectionName[0]);
+  aHitsCollections->AddHitsCollection(G4SDManager::GetSDMpointer()->GetCollectionID(m_calorimeterCollection), m_calorimeterCollection);
+}
+
+bool SimpleCalorimeterSDWithBirksLaw::ProcessHits(G4Step* aStep, G4TouchableHistory*)
+{
+  // check if energy was deposited
+  G4double edep = aStep->GetTotalEnergyDeposit();
+  if(edep==0.)
+    return false;
+
+  const G4String myMaterial = "Polystyrene";
+  const G4double birk1 = 0.0130*CLHEP::g/(CLHEP::MeV*CLHEP::cm2);
+  const G4double birk2 = 9.6e-6*CLHEP::g/(CLHEP::MeV*CLHEP::cm2)*CLHEP::g/(CLHEP::MeV*CLHEP::cm2);
```

Comment:
Sorry if that wasn't clear. I meant to add `myMaterial`, `birk1` and `birk2` as member variables to the class `SimpleCalorimeterSDWithBirksLaw`.